### PR TITLE
[extractor/moviepilot] Fix extractor

### PR DESCRIPTION
**Pull request by panatexxa**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixing extraction of Moviepilot trailers. The website content was changed.

Unfortunately, some keys are not available anymore. They are now coming from the Dailymotion extractor, I guess.
Because of the changed extraction, some other metadata are trivial to extract now.
However, they belong to the original movie, not the trailer, therefore I have sketched them as commented lines.
If you think, these metadata can be embedded in the video, I will update the extractor accordingly otherwise I will remove the lines.

Fixex (no bug assigned).

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/moviepilot.py [hunk 1]
```diff
@@ -91,22 +86,21 @@ def _real_extract(self, url):
...
-                     self._html_search_meta('duration', webpage, fatal=False) or ''),
-            lambda mobj: sum(float(x) * y for x, y in zip(mobj.groups(), (3600, 60, 1))))
-        # _html_search_meta is not used since we don't want name=description to match
-        description = self._html_search_regex(
-            '<meta[^>]+itemprop="description"[^>]+content="([^>"]+)"', webpage, 'description', fatal=False)
+        clip_store = self._search_json(
+            r'<script id="__NEXT_DATA__" type="application/json">',
+            webpage, 'clip store', video_id, fatal=True)
+        clip = traverse_obj(clip_store, ('props', 'initialProps', 'pageProps'),
+                            expected_type=dict) or {}
```
Comment (pukkandan) [inline yt_dlp/extractor/moviepilot.py:93]: It doesnt make sense for this to be non-fatal

File: yt_dlp/extractor/moviepilot.py [hunk 2]
```diff
@@ -91,22 +86,21 @@ def _real_extract(self, url):
...
+
+        # poster = traverse_obj(clip, ('poster', 'url'), expected_type=str) or ""
+        # movie_date = traverse_obj(clip, ('dataLayer', 'releasedata'), expected_type=str) or ""
+        # movie_rating = traverse_obj(clip, ('dataLayer', 'ageRestriction'), expected_type=int) or 0
```
Comment (pukkandan) [inline yt_dlp/extractor/moviepilot.py:97]: remove

File: yt_dlp/extractor/moviepilot.py [hunk 3]
```diff
@@ -91,22 +86,21 @@ def _real_extract(self, url):
...
             'ie_key': DailymotionIE.ie_key(),
             'display_id': video_id,
-            'title': self._og_search_title(webpage),
-            'url': self._html_search_meta('embedURL', webpage),
-            'thumbnail': self._html_search_meta('thumbnailURL', webpage),
-            'description': description,
-            'duration': duration,
-            'timestamp': parse_iso8601(self._html_search_meta('uploadDate', webpage), delimiter=' ')
+            'title': clip.get('title', ''),
+            'url': "https://www.dailymotion.com/video/" + clip['videoRemoteId'],
```
Comment (pukkandan) [inline yt_dlp/extractor/moviepilot.py:104]:
```suggestion
            'title': clip.get('title'),
            'url': f'https://www.dailymotion.com/video/{clip["videoRemoteId"]}',
```

File: yt_dlp/extractor/moviepilot.py [hunk 4]
```diff
@@ -91,22 +86,21 @@ def _real_extract(self, url):
...
+            'description': clip.get('summary', ''),
```
Comment (pukkandan) [inline yt_dlp/extractor/moviepilot.py:105]:
```suggestion
            'description': clip.get('summary'),
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/moviepilot.py [hunk 5]
```diff
@@ -91,22 +86,21 @@ def _real_extract(self, url):
...
-        duration = try_get(
-            re.match(r'P(?P<hours>\d+)H(?P<mins>\d+)M(?P<secs>\d+)S',
-                     self._html_search_meta('duration', webpage, fatal=False) or ''),
-            lambda mobj: sum(float(x) * y for x, y in zip(mobj.groups(), (3600, 60, 1))))
-        # _html_search_meta is not used since we don't want name=description to match
-        description = self._html_search_regex(
-            '<meta[^>]+itemprop="description"[^>]+content="([^>"]+)"', webpage, 'description', fatal=False)
+        clip_store = self._search_json(
+            r'<script id="__NEXT_DATA__" type="application/json">',
+            webpage, 'clip store', video_id, fatal=True)
```
[the remark this answers is not in the document]
Comment (panatexxa) [reply yt_dlp/extractor/moviepilot.py:91]: I doubt that this function works. I tried around a little but did not get it. The website does not contain the string "nuxt" and the regex within the function seems not to fit.

The relevant code on the Moviepilot website is:
```
<script id="__NEXT_DATA__" type="application/json">{"props":{"isServer":true,
"initialState":{"advertising":{"placements":{}},"comments":{},"consent":{},
"consumptions":{},"dashboard":{},"diggs":{},"followships":{},...
```

File: yt_dlp/extractor/moviepilot.py (hunk 1 above)
Comment (panatexxa) [reply yt_dlp/extractor/moviepilot.py]: I have replaced it now with a direct dict access and explicit error message. Should I further use `traverse_obj`?

File: yt_dlp/extractor/moviepilot.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/extractor/moviepilot.py:91]: sorry, I meant _search_nextjs_data

File: yt_dlp/extractor/moviepilot.py [hunk 6]
```diff
@@ -91,22 +86,22 @@ def _real_extract(self, url):
...
+
+        try:
+            clip = clip_store['props']['initialProps']['pageProps']
+        except KeyError as err:
+            raise ExtractorError(
+                'Unable to extract "clip" - Unexpected JSON',
+                cause=err, video_id=video_id)
```
Comment (pukkandan) [inline yt_dlp/extractor/moviepilot.py:98]:
```suggestion
        clip = clip_store['props']['initialProps']['pageProps']
```

File: yt_dlp/extractor/moviepilot.py [hunk 7]
```diff
@@ -91,22 +86,17 @@ def _real_extract(self, url):
...
-                     self._html_search_meta('duration', webpage, fatal=False) or ''),
-            lambda mobj: sum(float(x) * y for x, y in zip(mobj.groups(), (3600, 60, 1))))
-        # _html_search_meta is not used since we don't want name=description to match
-        description = self._html_search_regex(
-            '<meta[^>]+itemprop="description"[^>]+content="([^>"]+)"', webpage, 'description', fatal=False)
+        clip_store = self._search_json(
+            r'<script id="__NEXT_DATA__" type="application/json">',
+            webpage, 'clip store', video_id, fatal=True)
+
+        clip = clip_store['props']['initialProps']['pageProps']
```
Comment (pukkandan) [inline yt_dlp/extractor/moviepilot.py:93]:
```suggestion
        clip = self._search_nextjs_data(webpage, video_id)['props']['initialProps']['pageProps']
```

File: yt_dlp/extractor/moviepilot.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (panatexxa) [reply yt_dlp/extractor/moviepilot.py]: Thanks for fixing it! It seems to work as expected.

**Pull request merged**